### DISCUSSION TITLE
Proper type annotations now missing from ValidationException/RequestValidationError after v0.100

### DISCUSSION THREAD

**Discussion by MarkusSintonen** [fastapi/fastapi#9907] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# Before v0.100 you could raise custom validation exceptions like below.
# Also the RequestValidationError arguments were clear how they should be used.
# That also supported Mypy for type checking of the arguments.
raise RequestValidationError(
    [ErrorWrapper(ValueError("Start time can not be after end time"), "query.start_time")]
)

# But now in v0.100 its very unclear how RequestValidationError should be used.
# Its __init__ definition is just sequence of Any so we have lost type checking here after v0.100.
class RequestValidationError(ValidationException):
    def __init__(self, errors: Sequence[Any], *, body: Any = None) -> None:
        ... rest of it ...
```

### Description

Could you add back the type annotations for `errors: Sequence[Any]` validation error arguments without `Any` there. Now in v0.100 type checking is lost and code doesn't document it self very well anymore.

### Operating System

macOS

### FastAPI Version

0.100.0

### Python Version

Python 3.11.4

**Comment by MarkusSintonen** [comment]
Happy to open a fix PR!

**Comment by MarkusSintonen** [comment]
Opened a fix PR: https://github.com/tiangolo/fastapi/pull/9916

**Comment by tiangolo** [comment]
Thanks! I never intended `RequestValidationError` to be created internally by users' apps, at most captured/logged. Could you help me understand the use case that needs you to create an instance in your code?

**Reply by MarkusSintonen** [reply]
@tiangolo we are using FastAPIs `exception_handlers` to capture `RequestValidationError` and format it into a custom error response that conforms to `application/problem+json` with some additional details. Thank you!

**Reply by tiangolo** [reply, marked as answer]
Ah, I see! So, the idea with `RequestValidationError` is that FastAPI would raise it for you, and you can capture it and extract the data if you need to return a different response. But it is not intended for you to modify and raise it.

In that case, it would be better to raise a custom exception and handle that. This way you also don't depend on something internal to FastAPI, you can define it any way you want: https://fastapi.tiangolo.com/tutorial/handling-errors/ :nerd_face:

**Reply by MarkusSintonen** [reply]
Thanks!

> But it is not intended for you to modify and raise it.

We do not necessarily modify it. We format the errors into a response we use for errors.

> In that case, it would be better to raise a custom exception and handle that.

Thats not possible with all the requests being validated with Pydantic models in FastAPI. FastAPI internally converts Pydantic validation errors into `RequestValidationError` which we need to then capture in `exception_handlers` to convert into the proper error responses.

So essentially doing this:

```python
from fastapi.responses import JSONResponse
from fastapi.exceptions import RequestValidationError
from pydantic_core import ErrorDetails
from starlette.requests import Request

class JSONProblemResponse(JSONResponse):
    media_type = "application/problem+json"

async def request_validation_exception_handler(request: Request, exc: RequestValidationError) -> JSONProblemResponse:
    errors: Sequence[ErrorDetails] = exc.errors()  # type of this is Sequence[Any] instead of the Sequence[ErrorDetails] it actually is

    return JSONProblemResponse(status_code=400, content=error_content(errors))
```

Why the type of `exc.errors` is given as Any?

Its content is documented here https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions but for some reason the type annotations are missing.

**Reply by tiborrr** [reply]
I am still curious about the answer to this question. `Why the type of exc.errors is given as Any?` @tiangolo can you give some insight into this?
